### PR TITLE
[CPT-221] Use `debug_traceBlock` for `/block`

### PR DESCRIPTION
**Pull request by dhrubabasu**

### CODE REVIEW

File: client/eth.go [hunk 1]
```diff
@@ -50,15 +47,35 @@ func (c *EthClient) TxPoolContent(ctx context.Context) (*TxPoolContent, error) {
 	return content, err
 }
 
-// TraceTransaction returns a transaction trace
```
Comment (learyce) [inline client/eth.go:53]: Why did you kill all the comments?

File: client/eth.go [hunk 2]
```diff
@@ -50,15 +47,35 @@ func (c *EthClient) TxPoolContent(ctx context.Context) (*TxPoolContent, error) {
...
 func (c *EthClient) TraceTransaction(ctx context.Context, hash string) (*Call, []*FlatCall, error) {
-	result := &Call{}
+	var result *Call
```
Comment (learyce) [inline client/eth.go:51]: What's the difference between these two ways to init in Golang?

File: service/service_block.go [hunk 3]
```diff
@@ -140,7 +140,12 @@ func (s *BlockService) BlockTransaction(
...
 	}
 
-	transaction, terr := s.fetchTransaction(ctx, tx, header)
+	trace, flattened, err := s.client.TraceTransaction(ctx, tx.Hash().String())
+	if err != nil {
+		return nil, wrapError(errClientError, err)
+	}
+
+	transaction, terr := s.fetchTransaction(ctx, tx, header, trace, flattened)
 	if err != nil {
```
Comment (learyce) [inline service/service_block.go:149]: Should this be terr ?

File: service/service_block.go [hunk 4]
```diff
@@ -156,11 +161,14 @@ func (s *BlockService) fetchTransactions(
 ) ([]*types.Transaction, *types.Error) {
 	transactions := []*types.Transaction{}
 
-	for _, tx := range block.Transactions() {
-		transaction, err := s.fetchTransaction(ctx, tx, block.Header())
+	trace, flattened, err := s.client.TraceBlockByHash(ctx, block.Hash().String())
+
+	for i, tx := range block.Transactions() {
+		transaction, terr := s.fetchTransaction(ctx, tx, block.Header(), trace[i], flattened[i])
 		if err != nil {
```
Comment (learyce) [inline service/service_block.go:168]: Should this be terr ?

Comment (learyce) [inline service/service_block.go:168]: Should this be terr ?

File: client/eth.go (hunk 1 above)
Comment (dhrubabasu) [reply client/eth.go:53]: These comments are pretty useless imo, reverted in 5d40801 (will address in another PR)

File: service/service_block.go (hunk 4 above)
Comment (dhrubabasu) [reply service/service_block.go]: Yes, fixed in 6f5d163

Comment (dhrubabasu) [reply service/service_block.go]: Yes, fixed in 6f5d163

File: service/service_block.go (hunk 3 above)
Comment (dhrubabasu) [reply service/service_block.go]: Yes, fixed in 6f5d163

**Review by learyce — DISMISSED** [review]

File: client/eth.go (hunk 2 above)
Comment (patrick-ogrady) [reply client/eth.go:55]: This won't work @dhrubabasu. This just creates an unallocated pointer (is `nil` not `&Call{}`)

Comment (dhrubabasu) [reply client/eth.go]: Addressed in a9525f4

Comment (patrick-ogrady) [reply client/eth.go]: should do `var result Call`

**Review by patrick-ogrady — APPROVED** [review]
LGTM

**Pull request merged**